### PR TITLE
add extraMetadata to profile interface [#794]

### PR DESCRIPTION
**Pull request by bdon**

### CODE REVIEW

File: planetiler-core/src/main/java/com/onthegomap/planetiler/archive/TileArchiveMetadata.java [hunk 1]
```diff
@@ -175,6 +177,14 @@ public TileArchiveMetadata withJson(TileArchiveMetadataJson json) {
...
    */
 
+  private static Map<String,String> mergeMaps(Map<String,String> m1, Map<String,String> m2) {
+    return Stream.concat(m1.entrySet().stream(), m2.entrySet().stream())
+      .collect(Collectors.toMap(
+        Map.Entry::getKey,
+        Map.Entry::getValue,
+        (value1, value2) -> value2));
+  }
+
```
Comment (msbarry) [inline planetiler-core/src/main/java/com/onthegomap/planetiler/archive/TileArchiveMetadata.java:187]: Might be slightly more straightforward to use putAll here, that makes it explicit which direction we are clobbering too:

```java
var result = new TreeMap<>(m1);
result.putAll(m2);
return result;
```

File: planetiler-core/src/main/java/com/onthegomap/planetiler/Profile.java [hunk 2]
```diff
@@ -157,6 +158,8 @@ default boolean isOverlay() {
     return false;
   }
 
+  default Map<String,String> extraMetadata() { return Map.of(); }
```
Comment (msbarry) [inline planetiler-core/src/main/java/com/onthegomap/planetiler/Profile.java:161]: I wonder if we should include `archive` in the name here to make it clear where the metadata is going?

Comment (bdon) [reply planetiler-core/src/main/java/com/onthegomap/planetiler/Profile.java]: changed to `extraArchiveMetadata`

File: planetiler-core/src/main/java/com/onthegomap/planetiler/archive/TileArchiveMetadata.java [hunk 3]
```diff
@@ -93,7 +96,7 @@ public TileArchiveMetadata(Profile profile, PlanetilerConfig config, List<LayerA
       config.minzoom(),
       config.maxzoom(),
       vectorLayers == null ? null : new TileArchiveMetadataJson(vectorLayers),
-      mapWithBuildInfo(),
+      mergeMaps(mapWithBuildInfo(),profile.extraArchiveMetadata()),
```
Comment (msbarry) [inline planetiler-core/src/main/java/com/onthegomap/planetiler/archive/TileArchiveMetadata.java:99]: I think I'd expect the profile to provide defaults, then args to layer on top of that, so let's either reverse this or the behavior of mergeMaps?

Comment (bdon) [reply planetiler-core/src/main/java/com/onthegomap/planetiler/archive/TileArchiveMetadata.java:99]: shouldn't `mapWithBuildInfo` just be the `planetiler:*` metadata? If I define those in my profile (or command line args passed into my profile), they ought to override the defaults, I don't know why someone would do that though.

Comment (msbarry) [reply planetiler-core/src/main/java/com/onthegomap/planetiler/archive/TileArchiveMetadata.java:99]: Oh right! Yeah this looks good then. The command-line args get copied in just above this.

File: planetiler-core/src/main/java/com/onthegomap/planetiler/archive/TileArchiveMetadata.java [hunk 4]
```diff
@@ -17,6 +17,9 @@
 import java.util.LinkedHashMap;
 import java.util.List;
 import java.util.Map;
+import java.util.TreeMap;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
```
Comment (msbarry) [inline planetiler-core/src/main/java/com/onthegomap/planetiler/archive/TileArchiveMetadata.java:22]: remove unused imports

**Review by msbarry — COMMENTED** [review]
looks good! let's just remove the unused imports then good to merge.

**Pull request merged**